understanding/sharding.md

```diff
@@ -2,29 +2,210 @@
+{% endhint %}
+
+### BroacastEval
+
+[`broadcastEval`](https://discord.js.org/#/docs/main/v12/class/ShardClientUtil?scrollTo=broadcastEval) evaluates the input in the context of each shard's Client(s). This is what you should use when you want to execute a method or process data on a shard and return the result. It's useful for getting information that isn't available through client properties and must instead be retrieved through the use of methods.
+
+Example:
+```javascript
+/*
+    Example of result of broadcastEval() on a bot with 4 servers split across 2 shards.
+    Assume this is being executed on shard 0, the first shard.
+*/
+
+// If we just map our guilds' members.cache.size, it will return
+// only the mapped members.size of the shard this is being run on.
+console.log(client.guilds.cache.map((guild) => guild.members.cache.size));
+//		[
+//			30,
+//			25
+//		],
+
+// If we would like to map our guilds' members.cache.size from our
+// servers on all of our shards, we must make use of broadcastEval().
+// Remember, this runs in the context of the client, so we refer to the
+// Client using "this".
+const res = await client.shard.broadcastEval('this.guilds.cache.map((guild) => guild.members.cache.size)');
+console.log(res);
+// 	[
+//		[	// shard 0
+//			30,
+//			25
+//		],
+//		[	// shard 1
+//			55,
+//			10
+//		]
+// 	]
+
+````
+
+Say you want to get a guild from your client. In a non-sharded environment, you would simply use `client.guilds.cache.get('ID')` or something of that nature and then carry on with your code. In this case however, it is possible that the guild you're trying to get *is not present on the shard*. In order to get the guild for use, you would then need to fetch it from whatever shard it is present on using `broadcastEval()`.
+
+Here's an example of a function that uses `broadcastEval()` to get a single guild no matter what shard it is present on:
+
+```javascript
+/*
+  	Example by ZiNc#2032
+  	The following code fetches a single guild from across shards.
+    
+    NOTE: Fetched guild's properties such as "Guild.members.cache" and "Guild.roles.cache" will
+    not be Managers or Collections; these properties will be arrays of snowflake IDs.
+  	
+  	discord.js version 12.x
+  	client = new discordjs.Client()
+
+  	returns Guild
+*/
+
+const getServer = async (guildID) => {
+    // try to get guild from all the shards
+    const req = await client.shard.broadcastEval(`this.guilds.cache.get("${guildID}")`);
+
+    // return Guild or null if not found
+    return req.find(res => !!res) || null;
+}
+```
+
+{% hint style="info" %}
+`broadcastEval()` will only return basic javascript objects, and will not return Discord.js structures such as Collections, Guild objects, Member object, and the likes. You will need to interact directly with the API or build these structures yourself after getting a response back from your `broadcastEval()`.
```

Comment:
I've love to see an example structure, as we've seen above with all the other code.